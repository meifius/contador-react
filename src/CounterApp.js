import React, {useState} from 'react';
import PropTypes from 'prop-types';

/**
 * CounterApp es un componente que posee UI/UX para manipular y ver
 * un contador.
 * @param {NUmber} value
 */
const CounterApp = ({value}) => {

  const [counter, setCounter] = useState(value);

  const handleAdd = () => setCounter( c => c + 1);

  const handleMinus = () => setCounter( c => c - 1 );

  const handleReset = () => setCounter( c => value );

  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>

      <button onClick={handleAdd}>1+</button>
      <button onClick={handleMinus}>1-</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
}

CounterApp.propTypes = {
  value : PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
  value : 10,
}

export default CounterApp;