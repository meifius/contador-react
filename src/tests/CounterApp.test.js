// Importar Bibliotecas
import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import CounterApp from '../CounterApp';

describe('Testeo del componente <CounterApp />', () => {

  let wrapper = shallow( <CounterApp /> );

  beforeEach( () => {
    wrapper = shallow( <CounterApp /> );
  } );
  

  test('Debe de mostrar <CounterApp /> correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('Debe de mostrar por defecto el valor de 100', () => {
    const value = 100;
    const wrapper = shallow( <CounterApp value={value}/> );
    const valor = wrapper.find('h2').text().trim();

    expect( valor ).toBe( `${value}` );
  });

  test('Debe de incrementar con el boton +1', () => {
    wrapper.find('button').at(0).simulate('click');
    const counterText = wrapper.find('h2').text().trim();
    expect( counterText ).toBe( '11' );
  });
  
  test('Debe de decrementar con el boton -1', () => {
    wrapper.find('button').at(1).simulate('click');
    const counterText = wrapper.find('h2').text().trim();
    expect( counterText ).toBe( '9' );
  });
  
  test('Debe de restear con el boton Reset', () => {
    const value = 100;
    const wrapper = shallow( <CounterApp value={value}/> );

    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    const counterText = wrapper.find('h2').text().trim();
    expect( counterText ).toBe( `${value}` );
  });
    

});
