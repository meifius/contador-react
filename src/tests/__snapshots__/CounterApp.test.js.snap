// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeo del componente <CounterApp /> Debe de mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    1+
  </button>
  <button
    onClick={[Function]}
  >
    1-
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
